packages/twenty-server/src/database/commands/upgrade-version/0-23/0-23-set-user-vars-accounts-to-reconnect.command.ts

```diff
@@ -0,0 +1,170 @@
+import { ConnectedAccountWorkspaceEntity } from 'src/modules/connected-account/standard-objects/connected-account.workspace-entity';
+import { AccountsToReconnectKeys } from 'src/modules/connected-account/types/accounts-to-reconnect-key-value.type';
+import { MessageChannelSyncStatus } from 'src/modules/messaging/common/standard-objects/message-channel.workspace-entity';
+
+interface SetUserVarsAccountsToReconnectCommandOptions {
+  workspaceId?: string;
+}
+
+@Command({
+  name: 'upgrade-0.23:set-user-vars-accounts-to-reconnect',
+  description: 'Set user vars accounts to reconnect',
+})
+export class SetUserVarsAccountsToReconnectCommand extends CommandRunner {
+  private readonly logger = new Logger(
+    SetUserVarsAccountsToReconnectCommand.name,
+  );
+  constructor(
+    private readonly typeORMService: TypeORMService,
+    private readonly dataSourceService: DataSourceService,
+    private readonly workspaceCacheVersionService: WorkspaceCacheVersionService,
+    private readonly billingSubscriptionService: BillingSubscriptionService,
+    private readonly twentyORMGlobalManager: TwentyORMGlobalManager,
+    private readonly accountsToReconnectService: AccountsToReconnectService,
+    @InjectRepository(KeyValuePair, 'core')
+    private readonly keyValuePairRepository: Repository<KeyValuePair>,
+  ) {
+    super();
+  }
+
+  @Option({
+    flags: '-w, --workspace-id [workspace_id]',
+    description: 'workspace id. Command runs on all workspaces if not provided',
+    required: false,
+  })
+  parseWorkspaceId(value: string): string {
+    return value;
+  }
+
+  async run(
+    _passedParam: string[],
+    options: SetUserVarsAccountsToReconnectCommandOptions,
+  ): Promise<void> {
+    let activeSubscriptionWorkspaceIds: string[] = [];
+
+    if (options.workspaceId) {
+      activeSubscriptionWorkspaceIds = [options.workspaceId];
+    } else {
+      activeSubscriptionWorkspaceIds =
+        await this.billingSubscriptionService.getActiveSubscriptionWorkspaceIds();
+    }
+
+    if (!activeSubscriptionWorkspaceIds.length) {
+      this.logger.log(chalk.yellow('No workspace found'));
+
+      return;
+    } else {
+      this.logger.log(
+        chalk.green(
+          `Running command on ${activeSubscriptionWorkspaceIds.length} workspaces`,
+        ),
+      );
+    }
+
+    // Remove all deprecated user vars
+    await this.keyValuePairRepository.delete({
+      type: KeyValuePairType.USER_VAR,
+      key: 'ACCOUNTS_TO_RECONNECT',
+    });
+
+    for (const workspaceId of activeSubscriptionWorkspaceIds) {
+      try {
+        const dataSourceMetadata =
+          await this.dataSourceService.getDataSourcesMetadataFromWorkspaceId(
+            workspaceId,
+          )[0];
+
+        const workspaceDataSource =
+          await this.typeORMService.connectToDataSource(dataSourceMetadata);
+
+        if (workspaceDataSource) {
```

Comment:
It always return true if you check connectToDataSource code because we fallback to the mainDataSource.

Reply:
Ah okay